src/gpodder/sync.py

```diff
@@ -130,14 +130,16 @@ def get_track_length(filename):
             return int(float(mplayer_output[mplayer_output.index('ID_LENGTH'):].splitlines()[0][10:]) * 1000)
         except:
             pass
-    else:
-        logger.info('Please install MPlayer for track length detection.')
 
-    try:
-        mp3file = eyed3.mp3.Mp3AudioFile(filename)
-        return int(mp3file.info.time_secs * 1000)
-    except Exception as e:
-        logger.warn('Could not determine length: %s', filename, exc_info=True)
+    if eyd3mp3_available:
+        try:
+            mp3file = eyed3.mp3.Mp3AudioFile(filename)
+            return int(mp3file.info.time_secs * 1000)
+        except:
+            pass
```

Comment:
better `except Exception`.
See https://moserei.de/2017/09/20/python-bare-except.html

Also I'd rather see the error if there is one, at the expanse of duplicating the log.warn

Reply:
Fixed. I also fixed the mplayer case, or should its except not use Exception?

I also moved the two logger lines inside an else on the eyed3 branch, is that what you meant? One issue I see is mplayer failing, showing an error that it couldn't detect length and then eyed3 silently getting length. Should the eyed3 try use a finally with a warning that it detected the length, so the user knows?

```
finally:
    if ...mplayer...:
        logger.warn('eyed3.mp3 successfully determined length: %s', filename)
```

Of course these warnings would only be seen by gpo users and anyone looking in logs...

Reply:
Perfect!

> Should the eyed3 try use a finally with a warning that it detected the length, so the user knows?

It would be nice, but maybe overkill. As you wish. Let me know if you want to leave it as is. I'll merge then.

Reply:
Added in second commit.